src/tree-internal-api.ts

```diff
@@ -0,0 +1,48 @@
+import { TreeService } from './tree.service';
+import { Tree } from './tree';
+import { TreeModel } from './tree.types';
+import { TreeInternalComponent } from './tree-internal.component';
+import { Subject, Observable } from 'rxjs/Rx';
+import { Injectable, Inject, ElementRef } from '@angular/core';
+import * as _ from 'lodash';
+
+export class TreeInternalAPI {
+  constructor (private treeInternalComponent: TreeInternalComponent) { }
+
+  public select(e: MouseEvent = new MouseEvent('click')): void {
+    if (this.treeInternalComponent && !this.treeInternalComponent['isSelected'] && typeof this.treeInternalComponent['onNodeSelected'] === 'function') {
+      this.treeInternalComponent['onNodeSelected'](e);
```

Comment:
why do you need square brackets in here? TypeScript watches after types and will correct you at compile time